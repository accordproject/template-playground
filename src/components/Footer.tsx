import React, { useState } from "react";
import { Layout, Row, Col, Typography, Space, Button, Image, Grid } from "antd";
import {
  GithubOutlined,
  XOutlined,
  DiscordFilled,
  LinkedinFilled,
  DownOutlined,
  UpOutlined,
} from "@ant-design/icons";
import FOOTER_SECTION from "../constants/content/footer.json";
import { FooterSection, FooterLink } from "../types/components/Footer.types";

const { Footer } = Layout;
const { Text, Link } = Typography;
const { useBreakpoint } = Grid;

const CustomFooter: React.FC = () => {
  const year = new Date().getFullYear();
  const screens = useBreakpoint();
  const [expanded, setExpanded] = useState(false);

  return (
    <Footer
      id="footer"
      style={{
        background: "#1b2540",
        color: "white",
        padding: "50px 50px 20px 50px",
      }}
    >
      <Row justify="space-between" align="middle" gutter={[16, 16]}>
        <Col xs={24} md={12}>
          <Space direction="vertical" size="middle">
            <Link href="https://www.accordproject.org" target="_blank">
              <Image
                src="/logo.png"
                alt="Template Playground"
                preview={false}
                style={{ height: "36px", maxWidth: "100%" }}
              />
            </Link>
            <Text style={{ color: "rgba(255, 255, 255, 0.65)" }}>
              The open-source smart legal contract stack
            </Text>
            <Link href="mailto:admin@accordproject.org">
              <Text strong style={{ color: "rgba(255, 255, 255, 0.65)" }}>
                admin@accordproject.org
              </Text>
            </Link>
            <Link href="https://discord.com/invite/Zm99SKhhtA" target="_blank">
              <Button
                size="large"
                style={{
                  padding: "5px 30px",
                  backgroundColor: "#19c6c7",
                  borderRadius: "5px",
                  color: "#050c40",
                  textAlign: "center",
                  border: "none",
                }}
              >
                Join
              </Button>
            </Link>
          </Space>
        </Col>

        <Col xs={24} md={12}>
          {!screens.md && (
            <Button
              type="text"
              onClick={() => setExpanded(!expanded)}
              style={{ color: "white", fontSize: "16px", marginBottom: "10px" }}
            >
              {expanded ? <UpOutlined /> : <DownOutlined />} Other Links
            </Button>
          )}

          {(screens.md || expanded) && (
            <Row justify="end" gutter={[16, 16]}>
              {FOOTER_SECTION.sections.map((section: FooterSection) => (
                <Col xs={24} sm={12} md={6} key={section.title}>
                  <Space direction="vertical" size="middle">
                    <Text
                      strong
                      style={{
                        color: "rgba(255, 255, 255, 0.65)",
                        fontSize: "11px",
                        letterSpacing: "0.1em",
                      }}
                    >
                      {section.title}
                    </Text>
                    {section.links.map((link: FooterLink) => (
                      <Link
                        href={link.href}
                        key={link.title}
                        style={{ color: "white", fontSize: "15px" }}
                      >
                        {link.title}
                      </Link>
                    ))}
                  </Space>
                </Col>
              ))}
            </Row>
          )}
        </Col>
      </Row>

      <Row justify="space-between" align="middle" style={{ marginTop: "40px" }}>
        <Col>
          <Text style={{ color: "rgba(255, 255, 255, 0.85)" }}>
            copyright Â© {year} accord project &bull;{" "}
            <Link
              strong
              href="https://accordproject.org/privacy"
              target="_blank"
              style={{ color: "rgba(255, 255, 255, 0.85)" }}
            >
              trademark policy
            </Link>{" "}
            &bull;{" "}
            <Link
              strong
              href="https://accordproject.org/brand-assets"
              target="_blank"
              style={{ color: "rgba(255, 255, 255, 0.85)" }}
            >
              brand assets
            </Link>
          </Text>
        </Col>

        <Col>
          <Space>
            <Link href="https://github.com/accordproject" target="_blank" style={{ color: "rgba(255, 255, 255, 0.85)" }}>
              <GithubOutlined style={{ fontSize: "17px" }} />
            </Link>
            <Link href="https://twitter.com/AccordHQ" target="_blank" style={{ color: "rgba(255, 255, 255, 0.85)" }}>
              <XOutlined style={{ fontSize: "17px" }} />
            </Link>
            <Link href="https://discord.com/invite/Zm99SKhhtA" target="_blank" style={{ color: "rgba(255, 255, 255, 0.85)" }}>
              <DiscordFilled style={{ fontSize: "17px" }} />
            </Link>
            <Link href="https://www.linkedin.com/company/accordproject/" target="_blank" style={{ color: "rgba(255, 255, 255, 0.85)" }}>
              <LinkedinFilled style={{ fontSize: "17px" }} />
            </Link>
          </Space>
        </Col>
      </Row>
    </Footer>
  );
};

export default CustomFooter;
