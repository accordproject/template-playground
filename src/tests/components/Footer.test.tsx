import { render } from "@testing-library/react";
import { describe, it, expect, vi, beforeAll } from "vitest";
import Footer from "@components/Footer";

vi.mock("../../public/logo.png", () => ({
  default: "logo.png",
}));

vi.mock("../constants/content/footer.json", () => ({
  sections: [
    {
      title: "ABOUT",
      links: [
        { title: "About the AP", href: "https://accordproject.org/about/" },
        {
          title: "FAQ",
          href: "https://accordproject.org/frequently-asked-questions/",
        },
        {
          title: "White Paper",
          href: "https://accordproject.org/whitepaper-2024/",
        },
      ],
    },
    {
      title: "COMMUNITY",
      links: [
        { title: "Contribute", href: "https://accordproject.org/contribute/" },
        {
          title: "Tech WG",
          href: "https://accordproject.org/working-groups/technology/",
        },
        {
          title: "Join Discord",
          href: "https://discord.com/invite/Zm99SKhhtA",
          external: true,
        },
      ],
    },
    {
      title: "PROJECTS",
      links: [
        {
          title: "Concerto",
          href: "https://accordproject.org/projects/concerto/",
        },
        {
          title: "VS Code Extension",
          href: "https://marketplace.visualstudio.com/items?itemName=accordproject.cicero-vscode-extension",
        },
        {
          title: "Markdown Transform",
          href: "https://github.com/accordproject/markdown-transform",
        },
      ],
    },
    {
      title: "RESOURCES",
      links: [
        {
          title: "Template Library",
          href: "https://templates.accordproject.org/",
        },
        {
          title: "Model Repository",
          href: "https://models.accordproject.org/",
        },
        { title: "Videos", href: "https://vimeo.com/accordproject" },
        {
          title: "GitHub",
          href: "https://github.com/accordproject",
        },
        { title: "Documentation", href: "https://docs.accordproject.org/" },
      ],
    },
  ],
}));

beforeAll(() => {
  window.matchMedia = vi.fn().mockImplementation(() => ({
    matches: false,
    addListener: vi.fn(),
    removeListener: vi.fn(),
  }));
});

describe("Footer", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
