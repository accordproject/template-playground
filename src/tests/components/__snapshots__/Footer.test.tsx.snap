// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > matches the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="ant-layout-footer css-dev-only-do-not-override-17a39f8"
    id="footer"
    style="background: rgb(27, 37, 64); color: white; padding: 50px 50px 20px 50px;"
  >
    <div
      class="ant-row ant-row-space-between ant-row-middle css-dev-only-do-not-override-17a39f8"
      style="margin: -8px -8px -8px -8px;"
    >
      <div
        class="ant-col ant-col-xs-24 ant-col-md-12 css-dev-only-do-not-override-17a39f8"
        style="padding: 8px 8px 8px 8px;"
      >
        <div
          class="ant-space css-dev-only-do-not-override-17a39f8 ant-space-vertical"
        >
          <div
            class="ant-space-item"
            style="margin-bottom: 16px;"
          >
            <a
              class="ant-typography css-dev-only-do-not-override-17a39f8"
              href="https://www.accordproject.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="ant-image css-dev-only-do-not-override-17a39f8"
              >
                <img
                  alt="Template Playground"
                  class="ant-image-img css-dev-only-do-not-override-17a39f8"
                  src="/logo.png"
                  style="height: 36px; max-width: 100%;"
                />
              </div>
            </a>
          </div>
          <div
            class="ant-space-item"
            style="margin-bottom: 16px;"
          >
            <span
              class="ant-typography css-dev-only-do-not-override-17a39f8"
              style="color: rgba(255, 255, 255, 0.65);"
            >
              The open-source smart legal contract stack
            </span>
          </div>
          <div
            class="ant-space-item"
            style="margin-bottom: 16px;"
          >
            <a
              class="ant-typography css-dev-only-do-not-override-17a39f8"
              href="mailto:admin@accordproject.org"
            >
              <span
                class="ant-typography css-dev-only-do-not-override-17a39f8"
                style="color: rgba(255, 255, 255, 0.65);"
              >
                <strong>
                  admin@accordproject.org
                </strong>
              </span>
            </a>
          </div>
          <div
            class="ant-space-item"
          >
            <a
              class="ant-typography css-dev-only-do-not-override-17a39f8"
              href="https://discord.com/invite/Zm99SKhhtA"
              rel="noopener noreferrer"
              target="_blank"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-default ant-btn-lg"
                style="padding: 5px 30px; background-color: rgb(25, 198, 199); border-radius: 5px; color: rgb(5, 12, 64); text-align: center;"
                type="button"
              >
                <span>
                  Join
                </span>
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-xs-24 ant-col-md-12 css-dev-only-do-not-override-17a39f8"
        style="padding: 8px 8px 8px 8px;"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-17a39f8 ant-btn-text"
          style="color: white; font-size: 16px; margin-bottom: 10px;"
          type="button"
        >
          <span
            aria-label="down"
            class="anticon anticon-down"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
          <span>
             Other Links
          </span>
        </button>
      </div>
    </div>
    <div
      class="ant-row ant-row-space-between ant-row-middle css-dev-only-do-not-override-17a39f8"
      style="margin-top: 40px;"
    >
      <div
        class="ant-col css-dev-only-do-not-override-17a39f8"
      >
        <span
          class="ant-typography css-dev-only-do-not-override-17a39f8"
          style="color: rgba(255, 255, 255, 0.85);"
        >
          copyright © 2025 accord project • 
          <a
            class="ant-typography css-dev-only-do-not-override-17a39f8"
            href="https://accordproject.org/privacy"
            rel="noopener noreferrer"
            style="color: rgba(255, 255, 255, 0.85);"
            target="_blank"
          >
            <strong>
              trademark policy
            </strong>
          </a>
           • 
          <a
            class="ant-typography css-dev-only-do-not-override-17a39f8"
            href="https://accordproject.org/brand-assets"
            rel="noopener noreferrer"
            style="color: rgba(255, 255, 255, 0.85);"
            target="_blank"
          >
            <strong>
              brand assets
            </strong>
          </a>
        </span>
      </div>
      <div
        class="ant-col css-dev-only-do-not-override-17a39f8"
      >
        <div
          class="ant-space css-dev-only-do-not-override-17a39f8 ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <a
              class="ant-typography css-dev-only-do-not-override-17a39f8"
              href="https://github.com/accordproject"
              rel="noopener noreferrer"
              style="color: rgba(255, 255, 255, 0.85);"
              target="_blank"
            >
              <span
                aria-label="github"
                class="anticon anticon-github"
                role="img"
                style="font-size: 17px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="github"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0138.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                  />
                </svg>
              </span>
            </a>
          </div>
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <a
              class="ant-typography css-dev-only-do-not-override-17a39f8"
              href="https://twitter.com/AccordHQ"
              rel="noopener noreferrer"
              style="color: rgba(255, 255, 255, 0.85);"
              target="_blank"
            >
              <span
                aria-label="x"
                class="anticon anticon-x"
                role="img"
                style="font-size: 17px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="x"
                  fill="currentColor"
                  fill-rule="evenodd"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M921 912L601.11 445.75l.55.43L890.08 112H793.7L558.74 384 372.15 112H119.37l298.65 435.31-.04-.04L103 912h96.39L460.6 609.38 668.2 912zM333.96 184.73l448.83 654.54H706.4L257.2 184.73z"
                  />
                </svg>
              </span>
            </a>
          </div>
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <a
              class="ant-typography css-dev-only-do-not-override-17a39f8"
              href="https://discord.com/invite/Zm99SKhhtA"
              rel="noopener noreferrer"
              style="color: rgba(255, 255, 255, 0.85);"
              target="_blank"
            >
              <span
                aria-label="discord"
                class="anticon anticon-discord"
                role="img"
                style="font-size: 17px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="discord"
                  fill="currentColor"
                  fill-rule="evenodd"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M811.15 87c51.16 0 92.41 41.36 94.85 90.03V960l-97.4-82.68-53.48-48.67-58.35-50.85 24.37 80.2H210.41c-51 0-92.41-38.74-92.41-90.06V177.21c0-48.67 41.48-90.1 92.6-90.1h600.3zM588.16 294.1h-1.09l-7.34 7.28c75.38 21.8 111.85 55.86 111.85 55.86-48.58-24.28-92.36-36.42-136.14-41.32-31.64-4.91-63.28-2.33-90 0h-7.28c-17.09 0-53.45 7.27-102.18 26.7-16.98 7.39-26.72 12.22-26.72 12.22s36.43-36.42 116.72-55.86l-4.9-4.9s-60.8-2.33-126.44 46.15c0 0-65.64 114.26-65.64 255.13 0 0 36.36 63.24 136.11 65.64 0 0 14.55-19.37 29.27-36.42-56-17-77.82-51.02-77.82-51.02s4.88 2.4 12.19 7.27h2.18c1.09 0 1.6.54 2.18 1.09v.21c.58.59 1.09 1.1 2.18 1.1 12 4.94 24 9.8 33.82 14.53a297.58 297.58 0 0065.45 19.48c33.82 4.9 72.59 7.27 116.73 0 21.82-4.9 43.64-9.7 65.46-19.44 14.18-7.27 31.63-14.54 50.8-26.79 0 0-21.82 34.02-80.19 51.03 12 16.94 28.91 36.34 28.91 36.34 99.79-2.18 138.55-65.42 140.73-62.73 0-140.65-66-255.13-66-255.13-59.45-44.12-115.09-45.8-124.91-45.8l2.04-.72zM595 454c25.46 0 46 21.76 46 48.41 0 26.83-20.65 48.59-46 48.59s-46-21.76-46-48.37c.07-26.84 20.75-48.52 46-48.52zm-165.85 0c25.35 0 45.85 21.76 45.85 48.41 0 26.83-20.65 48.59-46 48.59s-46-21.76-46-48.37c0-26.84 20.65-48.52 46-48.52z"
                  />
                </svg>
              </span>
            </a>
          </div>
          <div
            class="ant-space-item"
          >
            <a
              class="ant-typography css-dev-only-do-not-override-17a39f8"
              href="https://www.linkedin.com/company/accordproject/"
              rel="noopener noreferrer"
              style="color: rgba(255, 255, 255, 0.85);"
              target="_blank"
            >
              <span
                aria-label="linkedin"
                class="anticon anticon-linkedin"
                role="img"
                style="font-size: 17px;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="linkedin"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1168.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
                  />
                </svg>
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
